catkin_tools/make_jobserver.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2014 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+from os import pipe, write, close, unlink, read
+from multiprocessing import cpu_count
+from tempfile import mkstemp
+from subprocess import call, PIPE
+from termios import FIONREAD
+
+import fcntl
+import array
+
+from .common import log
+
+FD_SUPPORT_MAKEFILE = b'''
+all:
+\techo $(MAKEFLAGS) | grep -- '--jobserver-fds'
+'''
+
+
+class MakeJobServer:
+
+    """
+    This class implements a GNU make job server.
+
+    TODO:
+     - use os.getloadavg() to maintain server load
```

Comment:
Ok, let's just remember to open an issue to track this if we merge this without addressing that.